dev/reference/best-practices/pd-scheduling.md

```diff
@@ -0,0 +1,270 @@
+    - The scheduling process is too slow. You can check the **Operator Step duration** metric to confirm the cause. Generally, steps that do not involve sending and receiving snapshots (such as `TransferLeader`, `RemovePeer`, `PromoteLearner`) should be completed in milliseconds, while steps that involve snapshots (such as `AddLearner` and `AddPeer`) are expected to be completed in tens of seconds. If the duration is obviously too long, it could be caused by high pressure on TiKV or bottleneck in network, etc., which needs specific analysis.
+
+- PD fails to generate the corresponding balancing Scheduler. Possible reasons include:
+
+    - The Scheduler is not activated. For example, the corresponding Scheduler is deleted, or its limit it set to "0".
+    - Other constraints. For example, `evict-leader-scheduler` in the system prevents Leaders from being migrating to the corresponding Store. Or Label property is set, which makes some Stores reject Leaders.
+    - Restrictions from the cluster topology. For example, in a cluster of 3 replicas across 3 data centers, 3 replicas of each Region are distributed in different data centers due to replica isolation. If the number of Stores is different among these data centers, the scheduling can only reach a balanced state within each data center, but not balanced globally.
+
+### Taking nodes offline is slow
+
+This scenario requires examining the generation and execution of Operators through related metrics.
+
+If Operators are successfully generated but the scheduling process is slow, possible reasons are:
+
+- The scheduling speed is limited by default. You can adjust `leader-schedule-limit` or `replica-schedule-limit` to larger value.s Similarly, you can consider loosening the limits on `max-pending-peer-count` and `max-snapshot-count`.
+- Other scheduling tasks are running concurrently and racing for resources in the system. You can refer to the solution in [the previous section](#leadersregions-are-not-evenly-distributed).
+- When you take a single node offline, a number of Region Leaders to be processed (around 1/3 under the configuration of 3 replicas) are distributed on the node to remove. Therefore, the speed is limited by the speed at which snapshots are generated by this single node. You can speed it up by manually adding an `evict-leader-scheduler` to migrate Leaders.
+
+If the corresponding Operator fails to generate, possible reasons are:
+
+- The Operator is stopped, or `replica-schedule-limit` is set to "0".
+- There is no proper node for Region migration. For example, if the available capacity size of the replacing nodes (of the same label) is less than 20%, PD will stop scheduling to avoid running out of storage space. In such case, you need to add more nodes or delete some data to free the space.
```

Comment:
```suggestion
- There is no proper node for Region migration. For example, if the available capacity size of the replacing node (of the same label) is less than 20%, PD will stop scheduling to avoid running out of storage on that node. In such case, you need to add more nodes or delete some data to free the space.
```